src/lib/db.ts

```diff
@@ -470,44 +472,24 @@ export class DB {
     });
   }
 
-  public publishDocumentAsSupport(documentModel: DocumentModelType,
-                                  audience: AudienceModelType,
-                                  sectionTarget: SectionTarget,
-                                  caption: string) {
-    const {user} = this.stores;
+  public publishDocumentAsSupport(documentModel: DocumentModelType, caption: string) {
+    const { user } = this.stores;
     const content = documentModel.content.publish();
-    return new Promise<{document: DBDocument, metadata: DBSupportPublicationMetadata}>((resolve, reject) => {
-      this.createDocument({ type: SupportPublication, content })
-        .then(({document, metadata}) => {
-          const supportPath = this.firebase.getSupportsPath(user, audience, sectionTarget);
-          const supportRef = this.firebase.ref(supportPath).push();
-          const fbSupportDocPath = this.firebase.getUserDocumentPath(user, document.self.documentKey);
-          const support: DBPublishedSupport = {
-            version: "1.0",
-            self: {
-              classHash: user.classHash,
-              offeringId: user.offeringId,
-              audienceType: audience.type,
-              audienceId: audience.identifier || "",
-              sectionTarget,
-              key: supportRef.key!
-            },
-            uid: user.id,
-            properties: { caption, ...documentModel.copyProperties() },
-            originDoc: documentModel.key,
-            timestamp: firebase.database.ServerValue.TIMESTAMP as number,
-            type: ESupportType.publication,
-            content: fbSupportDocPath,
-            deleted: false
-          };
-
-          supportRef.set(support)
-            .then(() => {
-              Logger.logDocumentEvent(LogEventName.PUBLISH_SUPPORT, documentModel);
-              resolve({document, metadata: metadata as DBSupportPublicationMetadata});
-            })
-            .catch(reject);
-        });
+    const fs = this.firestore;
+    return fs.batch(batch => {
+      const rootRef = fs.documentRef(fs.getRootFolder());
+      batch.set(rootRef, { updatedAt: fs.timestamp() });
+      const docRef = fs.newDocumentRef(fs.getMulticlassSupportsPath());
+      batch.set(docRef, {
+        uid: user.id,
+        type: "supportPublication",
+        createdAt: fs.timestamp(),
+        properties: { teacherSupport: "true", caption, ...documentModel.copyProperties() },
+        problem: getProblemPath(this.stores),
```

Comment:
`activity-player` uses the offering ID for "problem" matching. Not clear to me a priori which approach is better.

Reply:
I went ahead and added some additional LTI-related fields in case they might prove useful down the road.